diff --git c/zos/ProcessField.h w/zos/ProcessField.h
new file mode 100644
index 0000000..71170e0
--- /dev/null
+++ w/zos/ProcessField.h
@@ -0,0 +1,25 @@
+#ifndef HEADER_ZOSProcessField
+#define HEADER_ZOSProcessField
+/*
+htop - zos/ProcessField.h
+(C) 2020 htop dev team
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+
+#define PLATFORM_PROCESS_FIELDS  \
+   ZONEID = 100,                 \
+   ZONE  = 101,                  \
+   PROJID = 102,                 \
+   TASKID = 103,                 \
+   POOLID = 104,                 \
+   CONTID = 105,                 \
+   LWPID = 106,                  \
+   ASID = 107,                   \
+   JOBNAME = 108,                \
+                                 \
+   DUMMY_BUMP_FIELD = CWD,
+
+
+#endif /* HEADER_ZOSProcessField */
