diff --git c/zos/ZOSMachine.h w/zos/ZOSMachine.h
new file mode 100644
index 0000000..6af45d0
--- /dev/null
+++ w/zos/ZOSMachine.h
@@ -0,0 +1,58 @@
+#ifndef HEADER_ZOSMachine
+#define HEADER_ZOSMachine
+/*
+htop - ZOSMachine.h
+(C) 2014 Hisham H. Muhammad
+(C) 2017,2018 Guy M. Broome
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <sys/param.h>
+#include <sys/resource.h>
+#include <sys/uio.h>
+
+#include "Machine.h"
+#include "Hashtable.h"
+#include "UsersTable.h"
+
+
+
+#define ZONE_ERRMSGLEN 1024
+extern char zone_errmsg[ZONE_ERRMSGLEN];
+
+typedef struct CPUData_ {
+   double userPercent;
+   double nicePercent;
+   double systemPercent;
+   double irqPercent;
+   double idlePercent;
+   double systemAllPercent;
+   double frequency;
+   uint64_t luser;
+   uint64_t lkrnl;
+   uint64_t lintr;
+   uint64_t lidle;
+   bool online;
+} CPUData;
+
+typedef struct ZOSMachine_ {
+   Machine super;
+
+   CPUData* cpus;
+
+   int pageSize;
+   int pageSizeKB;
+
+   uint64_t globalUserTime;
+   uint64_t globalNiceTime;
+   uint64_t globalSysTime;
+   uint64_t prevGlobalUserTime;
+   uint64_t prevGlobalNiceTime;
+   uint64_t prevGlobalSysTime;
+   uint64_t prevTime;
+} ZOSMachine;
+
+#endif
