diff --git c/CRT.c w/CRT.c
index 9c8b841..8252a10 100644
--- c/CRT.c
+++ w/CRT.c
@@ -71,7 +71,7 @@ static const char* const CRT_treeStrAscii[LAST_TREE_STR] = {
    [TREE_STR_DESC] = "-",
 };
 
-#ifdef HAVE_LIBNCURSESW
+#if defined(HAVE_LIBNCURSESW) || defined(__MVS__)
 
 static const char* const CRT_treeStrUtf8[LAST_TREE_STR] = {
    [TREE_STR_VERT] = "\xe2\x94\x82", // │
@@ -97,7 +97,7 @@ static const Settings* CRT_settings;
 const char* CRT_degreeSign;
 
 static const char* initDegreeSign(void) {
-#ifdef HAVE_LIBNCURSESW
+#if defined(HAVE_LIBNCURSESW) || defined(__MVS__)
    if (CRT_utf8)
       return "\xc2\xb0";
 
@@ -1127,18 +1127,26 @@ IGNORE_WCASTQUAL_END
       CRT_colorScheme = COLORSCHEME_MONOCHROME;
    CRT_setColors(CRT_colorScheme);
 
-#ifdef HAVE_LIBNCURSESW
-   if (allowUnicode && String_eq(nl_langinfo(CODESET), "UTF-8")) {
+#if defined(HAVE_LIBNCURSESW) || defined(__MVS__)
+#ifdef __MVS__
+   if (allowUnicode) {
       CRT_utf8 = true;
    } else {
       CRT_utf8 = false;
    }
+#else
+   if (allowUnicode && (String_eq(nl_langinfo(CODESET), "UTF-8"))) {
+      CRT_utf8 = true;
+   } else {
+      CRT_utf8 = false;
+   }
+#endif
 #else
    (void) allowUnicode;
 #endif
 
    CRT_treeStr =
-#ifdef HAVE_LIBNCURSESW
+#if defined(HAVE_LIBNCURSESW) || defined(__MVS__)
       CRT_utf8 ? CRT_treeStrUtf8 :
 #endif
       CRT_treeStrAscii;
