diff --git a/Meter.c b/Meter.c
index 86f8885..91a26af 100644
--- a/Meter.c
+++ b/Meter.c
@@ -320,6 +320,7 @@ static void GraphMeterMode_draw(Meter* this, int x, int y, int w) {
       return;
 
    const Machine* host = this->host;
+#ifndef __MVS__
    if (!timercmp(&host->realtime, &(data->time), <)) {
       int globalDelay = host->settings->delay;
       struct timeval delay = { .tv_sec = globalDelay / 10, .tv_usec = (globalDelay % 10) * 100000L };
@@ -329,6 +330,7 @@ static void GraphMeterMode_draw(Meter* this, int x, int y, int w) {
 
       data->values[nValues - 1] = sumPositiveValues(this->values, this->curItems);
    }
+#endif
 
    if (w <= 0)
       return;
