diff --git a/zos/openzfs_sysctl.h b/zos/openzfs_sysctl.h
new file mode 100644
index 0000000..27fa720
--- /dev/null
+++ b/zos/openzfs_sysctl.h
@@ -0,0 +1,17 @@
+#ifndef HEADER_openzfs_sysctl
+#define HEADER_openzfs_sysctl
+/*
+htop - generic/openzfs_sysctl.h
+(C) 2014 Hisham H. Muhammad
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "zfs/ZfsArcStats.h"
+
+
+void openzfs_sysctl_init(ZfsArcStats* stats);
+
+void openzfs_sysctl_updateArcStats(ZfsArcStats* stats);
+
+#endif
