diff --git a/zos/ZOSProcess.c b/zos/ZOSProcess.c
new file mode 100644
index 00000000..e7ff91a7
--- /dev/null
+++ b/zos/ZOSProcess.c
@@ -0,0 +1,31 @@
+#include "zos/ZOSProcess.h"
+
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "CRT.h"
+#include "Macros.h"
+#include "Process.h"
+#include "ProvideCurses.h"
+
+/* semi-global */
+int pageSize;
+int pageSizeKB;
+
+const ProcessFieldData Process_fields[LAST_PROCESSFIELD] = {
+   [0] = { .name = "", .title = NULL, .description = NULL, .flags = 0, },
+   [PID] = { .name = "PID", .title = "PID", .description = "Process/thread ID", .flags = 0, .pidColumn = true, },
+   [COMM] = { .name = "Command", .title = "Command ", .description = "Command line", .flags = 0, },
+   [PPID] = { .name = "PPID", .title = "PPID", .description = "Parent process ID", .flags = 0, .pidColumn = true, }
+};
+
+void Process_delete(Object* cast) {
+   ZOSProcess* this = (ZOSProcess*) cast;
+   Process_done((Process*)cast);
+   free(this->cgroup_short);
+   free(this->cgroup);
+   free(this->secattr);
+   free(this);
+}
